examples/renderers/NMR/example3.py

```diff
@@ -0,0 +1,227 @@
+
+        image_ref = torch.from_numpy(imread(image_path).astype(
+            'float32') / 255.).permute(2, 0, 1)[:3, ...][None, ::]
+        self.register_buffer('image_ref', image_ref)
+
+        ###########################
+        # Setup renderer
+        ###########################
+
+        renderer = Renderer(camera_mode='look_at')
+        # renderer.perspective = False
+        renderer.light_intensity_directional = 0.0
+        renderer.light_intensity_ambient = 1.0
+        self.renderer = renderer
+
+    def forward(self):
+        ###########################
+        # Render
+        ###########################
+
+        self.renderer.eye = get_points_from_angles(
+            CAMERA_DISTANCE, ELEVATION,
+            np.random.uniform(0, 360)
+        )
+        image, _, _ = self.renderer(
+            self.vertices,
+            self.faces,
+            torch.tanh(self.textures)
+        )
+        loss = torch.sum((image - self.image_ref) ** 2)
+
+        return loss
+
+
+def main():
+    args = parse_arguments()
+
+    ###########################
+    # Setup model
+    ###########################
+
+    model = Model(args.mesh, args.image)
+    model.cuda()
+
+    ###########################
+    # Optimize
+    ###########################
+
+    loop = tqdm.tqdm(range(args.epochs))
+    loop.set_description('Optimizing')
+
+    optimizer = torch.optim.Adam(
+        filter(lambda p: p.requires_grad, model.parameters()),
+        lr=0.1, betas=(0.5, 0.999)
+    )
+
+    azimuth = 0.0
+
+    os.makedirs(args.output_path, exist_ok=True)
+    writer = imageio.get_writer(os.path.join(
+        args.output_path, 'example3_optimization.gif'), mode='I')
+    for i in loop:
+        optimizer.zero_grad()
+
+        loss = model()
+
+        loss.backward()
+        optimizer.step()
+
+        model.renderer.eye = get_points_from_angles(
+            CAMERA_DISTANCE, ELEVATION, azimuth)
+
+        images, _, _ = model.renderer(
+            model.vertices,
+            model.faces,
+            torch.tanh(model.textures)
+        )
+
+        image = images.detach().cpu().numpy()[0].transpose(
+            (1, 2, 0))
+        writer.append_data((255 * image).astype(np.uint8))
+
+        azimuth = (azimuth + 4) % 360
+
+    writer.close()
+
+    ###########################
+    # Render optimized mesh
+    ###########################
+
+    loop = tqdm.tqdm(range(0, 360, 4))
+    loop.set_description('Drawing')
+
+    os.makedirs(args.output_path, exist_ok=True)
+    writer = imageio.get_writer(os.path.join(
+        args.output_path, 'example3_mesh.gif'), mode='I')
+    for azimuth in loop:
+        model.renderer.eye = get_points_from_angles(
+            CAMERA_DISTANCE, ELEVATION, azimuth)
+
+        images, _, _ = model.renderer(
+            model.vertices,
+            model.faces,
+            torch.tanh(model.textures)
+        )
+
+        image = images.detach().cpu().numpy()[0].transpose(
```

Comment:
May not be a big deal but you can use `permute` on gpu before transfering to cpu